src/main/java/net/sf/mzmine/modules/peaklistmethods/isotopes/isotopepeakscanner/IsotopePeakScannerTask.java

```diff
@@ -404,7 +404,8 @@ public void run() {
         // The results you miss by skipping here would have not been valid results anyway, so this
         // is not urgent. Will be nicer though, because of cleaner code.
 
-      PeakListRow parent = copyPeakRow(peakList.getRow(i));
+//      PeakListRow parent = copyPeakRow(peakList.getRow(i));
+      PeakListRow parent = copyPeakRow(plh.getRowByID(candidates[bestPatternIndex].get(0).getCandID()));

```

Comment:
This kind of statement is bad practice, because it contains multiple points where it may throw a NullPointerException or ArrayIndexOutOfBoundsException. Please split this into multiple lines, with each line performing a single operation. Also, add proper checks that the variables don't have any unexpected values (like null).

Reply:
Done. Generally, the results are checked before, if they are ok, else it doesn't get to this point.
But I also think, that the code of module is a bit messy. After i resolved the DataPointProcessing stuff, my goal is to rewrite this and make it cleaner and easier understandable and more functional.